### PR TITLE
Bug fix in Service Endpoint policies

### PR DESCRIPTION
**Pull request by chandrasekarsrinivasan**
## Checklist

- [ ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/Network/Commands.Network/ServiceEndpointPolicy/ServiceEndpointPolicyDefinition/GetAzureServiceEndpointPolicyDefinitionCommand.cs [hunk 1]
```diff
@@ -34,23 +34,10 @@ public class GetAzureServiceEndpointPolicyDefinitionCommand : NetworkBaseCmdlet
             HelpMessage = "The Service endpoint policy")]
         public PSServiceEndpointPolicy ServiceEndpointPolicy { get; set; }
 
-        [Parameter(
-            Mandatory = true,
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network/ServiceEndpointPolicy/ServiceEndpointPolicyDefinition/GetAzureServiceEndpointPolicyDefinitionCommand.cs]: This is a breaking change (removing amandatory param)... why are you pulling out the get by resourceID?

Comment (chandrasekarsrinivasan) [reply src/ResourceManager/Network/Commands.Network/ServiceEndpointPolicy/ServiceEndpointPolicyDefinition/GetAzureServiceEndpointPolicyDefinitionCommand.cs]: this was a bug. This should not be there in first place. There is no server call to query with resourceId. We cannot have this sort of a query for this API.

Comment (praries880) [reply src/ResourceManager/Network/Commands.Network/ServiceEndpointPolicy/ServiceEndpointPolicyDefinition/GetAzureServiceEndpointPolicyDefinitionCommand.cs]: @chandrasekarsrinivasan 
The Cmdlet is not using the resourceId anyways, 
It is [pulling out the name from the resourceId](https://github.com/Azure/azure-powershell/blob/f6a2882e0aa408b44465aca4226021547fa31be7/src/ResourceManager/Network/Commands.Network/ServiceEndpointPolicy/ServiceEndpointPolicyDefinition/GetAzureServiceEndpointPolicyDefinitionCommand.cs#L51) and using that to [make the call](https://github.com/Azure/azure-powershell/blob/f6a2882e0aa408b44465aca4226021547fa31be7/src/ResourceManager/Network/Commands.Network/ServiceEndpointPolicy/ServiceEndpointPolicyDefinition/GetAzureServiceEndpointPolicyDefinitionCommand.cs#L62 into the service. 

Can you elaborate on what the bug is?

**Comment by praries880** [comment]
@chandrasekarsrinivasan  As per discussion
* Mark the param ResorceId as non mandatory and use the [param deprecation attribute](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-attribute-help.md#a-parameter-is-being-deprecated) on it.

**Comment by praries880** [comment]
ping @chandrasekarsrinivasan

**Comment by EvgenyAgafonchikov** [comment]
CI build fixed. @praries880, please have a look

File: src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/VirtualNetworkTests.ps1 [hunk 2]
```diff
@@ -802,6 +803,14 @@ function Test-VirtualNetworkSubnetServiceEndpointPolicies
         Assert-AreEqual $serviceEndpointPolicyDefinitionName $getserviceEndpointPolicy[0].ServiceEndpointPolicyDefinitions[0].Name;
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/VirtualNetworkTests.ps1]: Is there an associated test recording that needs to be updated for this?

**Pull request merged**